public/javascripts/store.js

```diff
@@ -0,0 +1,39 @@
+//vuex state store to share state between all components
+function makeStore() {
+  var leftPanelClass = 'cc-leftPanel-mobile-hide',
+      userInfoDefault = {
+        email: '',
+        imgSrc: 'https://s.ytimg.com/yts/img/avatar_720-vflYJnzBZ.png'
+      },
+      userInfo = userInfoDefault;
+  if (firebase.auth().currentUser) {
+    var user = firebase.auth().currentUser;
+    user.imgSrc = user.photoURL || 'https://s.ytimg.com/yts/img/avatar_720-vflYJnzBZ.png';
+    userInfo = user;
+  }
+  return {
+    state: {
+      leftPanelClass,
+      userInfo
+    },
+    mutations: {
+      toggleState: function(state, newState, property) {
```

Comment:
seems like we should be more specific about what attributes of the state we are toggling - do we need a general function like this? Could it be dangerous?